zebra/zebra_srv6_vty.c

```diff
@@ -259,6 +260,129 @@ DEFUN (show_srv6_locator_detail,
+					       ", %s(%d)",
+					       zebra_route_string(client->proto),
+					       client->instance);
+
+			if (ret > 0)
+				i += ret;
+		}
+	}
+
+	/* Behavior */
+	if (sid->locator) {
+		if ((sid->locator->sid_format &&
+		     sid->locator->sid_format->type ==
+			     SRV6_SID_FORMAT_TYPE_USID) ||
+		    (!sid->locator->sid_format &&
+		     CHECK_FLAG(sid->locator->flags, SRV6_LOCATOR_USID))) {
+			switch (sid->ctx->ctx.behavior) {
+			case ZEBRA_SEG6_LOCAL_ACTION_END:
+				snprintf(behavior, sizeof(behavior), "uN");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_X:
+				snprintf(behavior, sizeof(behavior), "uA");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DX6:
+				snprintf(behavior, sizeof(behavior), "uDX6");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DX4:
+				snprintf(behavior, sizeof(behavior), "uDX4");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DT6:
+				snprintf(behavior, sizeof(behavior), "uDT6");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DT4:
+				snprintf(behavior, sizeof(behavior), "uDT4");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DT46:
+				snprintf(behavior, sizeof(behavior), "uDT46");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_UNSPEC:
+				snprintf(behavior, sizeof(behavior), "unspec");
+				break;
+			case ZEBRA_SEG6_LOCAL_ACTION_END_T:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_DX2:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_B6:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_B6_ENCAP:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_BM:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_S:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_AS:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_AM:
+			case ZEBRA_SEG6_LOCAL_ACTION_END_BPF:
+			default:
+				snprintf(behavior, sizeof(behavior), "unknown");
+			}
+		} else {
+			snprintf(behavior, sizeof(behavior), "%s",
+				 seg6local_action2str(sid->ctx->ctx.behavior));
+		}
+	}
+
+	/* SID context */
+	switch (sid->ctx->ctx.behavior) {
+	case ZEBRA_SEG6_LOCAL_ACTION_END:
+		break;
+	case ZEBRA_SEG6_LOCAL_ACTION_END_X:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DX6:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DX4:
+		RB_FOREACH (vrf, vrf_id_head, &vrfs_by_id) {
+			ifp = if_lookup_by_index(sid->ctx->ctx.ifindex,
+						 vrf->vrf_id);
+			if (ifp)
+				snprintf(ctx, sizeof(ctx), "Interface '%s'",
+					 ifp->name);
+		}
+		break;
+	case ZEBRA_SEG6_LOCAL_ACTION_END_T:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DT6:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DT4:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DT46:
+		vrf = vrf_lookup_by_id(sid->ctx->ctx.vrf_id);
+		snprintf(ctx, sizeof(ctx), "VRF '%s'",
+			 vrf ? vrf->name : "<unknown>");
+		break;
+	case ZEBRA_SEG6_LOCAL_ACTION_END_DX2:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_B6:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_B6_ENCAP:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_BM:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_S:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_AS:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_AM:
+	case ZEBRA_SEG6_LOCAL_ACTION_END_BPF:
+	case ZEBRA_SEG6_LOCAL_ACTION_UNSPEC:
+	default:
```

Comment:
again, please remove the default case.  If someone adds a new local action, I want the compile to fail and the person adding the new action be responsible for fixing it.